spec/event_store_client/data_decryptor_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'securerandom'
+require 'support/dummy_repository'
+
+module EventStoreClient
+  RSpec.describe DataDecryptor do
+    describe '#call' do
+      let(:repository) { DummyRepository.new }
+
+      subject { described_class.new(data: data, schema: schema, repository: repository) }
+
+      it 'returns decrypted data' do
+        expect(subject.call).to eq(
+          user_id: user_id,
+          first_name: 'Anakin',
+          last_name: 'Skylwalker',
+          profession: 'Jedi'
+        )
+      end
+
+      it 'updates the data reader' do
+        subject.call
+        expect(subject.decrypted_data).to eq(
+          user_id: user_id,
+          first_name: 'Anakin',
+          last_name: 'Skylwalker',
+          profession: 'Jedi'
+        )
+      end
+
+      it 'skips the decryption of non-existing keys' do
+        schema[:attributes] << :side
+        expect(subject.call).to eq(
+          user_id: user_id,
+          first_name: 'Anakin',
+          last_name: 'Skylwalker',
+          profession: 'Jedi'
+        )
+      end
+    end
+
+    let(:key_repository) { DummyRepository.new }
```

Comment:
Not important but, I always keep let declarations at the beginning of the tests suite, or as close as I need the value if I have for example several contexts(so put specific let inside the context and common at the beginning of specs).  But its only my approach.

Reply:
It makes no big difference if test files are kept small, and such as class files.

For me it's nice to have a list of tests in the file first - so I know better what the test does.
When you have small files, you don't need to jump often and work on tests. I've found tests more used to get an overview of what the class actually does. :D We can discuss it more and normalize it somehow. I don't mind having declarations at the beginning.

Reply:
Yes it was only my thought